feathr_project/feathrcli/data/feathr_user_workspace/features/request_features.py

```diff
@@ -0,0 +1,36 @@
+from feathr.anchor import FeatureAnchor
+from feathr.feature import Feature
+from feathr.dtype import BOOLEAN, INT32, FLOAT, ValueType
+from feathr.feature_derivations import DerivedFeature
+from feathr.source import PASSTHROUGH_SOURCE
+
+f_trip_distance = Feature(name="f_trip_distance", feature_type=FLOAT, transform="trip_distance")
+f_trip_time_duration = Feature(name="f_trip_time_duration",
+            feature_type=INT32,
+            transform="time_duration(lpep_pickup_datetime, lpep_dropoff_datetime, 'minutes')")
+
+features = [
+    f_trip_distance,
+    f_trip_time_duration,
+    Feature(name="f_is_long_trip_distance",
+            feature_type=BOOLEAN,
+            transform="cast_float(trip_distance)>30"),
+    Feature(name="f_day_of_week",
+            feature_type=INT32,
+            transform="dayofweek(lpep_dropoff_datetime)"),
+  ]
+
+request_anchor = FeatureAnchor(name="request_features",
```

Comment:
I’m unsure about the idiom we are using here. Wouldn’t the feature definition script look more intuitive if each “passthrough” `Feature` here was self-contained? User may wonder why they need to invoke `FeatureAnchor` here, and why they can’t just specify the `Feature`s and be done with it.
Something that looks “off” about this IMO is that the expressions like `transform=“trip_distance”` are isolated from the context about what `source` is actually being used, even though they can’t really be understood without that context.
One possible “fix” might just be via style recommendation, e.g. we can encourage users to invoke all those `Feature` constructors etc. inline here within the `FeatureAnchor` constructor invocation (which incidentally would look more like Frame HOCON). If we want to do that, we should adopt this style in examples like this, and in the docs.

Reply:
Maybe a crazy thought is to just attach source to the feature level? And hide the FeatureAnchor concept completely?

When building the config file in `RepoDefinitions`, we can then just auto group features in the same source together and synthesize the feature anchor there?

I know this is slightly different from the HOCON concept where we do feature group authoring as the level of atomicity but I think this makes sense to do so as a declarative API to attach the same source to multiple features and "optimize" it by grouping common source things together as part of the config compilation step.

Reply:
+1 to Ben's idea about attaching it to the feature level. @jaymo001 WDYT?